packages/concerto-core/lib/codegen/javascriptparser.js

```diff
@@ -251,12 +251,17 @@ class JavaScriptParser {
                                 throws = JavaScriptParser.getThrows(comment);
                                 example = JavaScriptParser.getExample(comment);
                             }
+                            let name = thing.key.name;
+                            if(!thing.key.name && thing.key.property.name){
+                                name = thing.key.property.name;
+                            }
+
                             commentData = commentData || [];
                             if (visibility === '+' || visibility === '~' || includePrivates) {
                                 const method = {
                                     visibility: visibility,
                                     returnType: returnType,
-                                    name: thing.key.name,
```

Comment:
I'm unclear about those changes, do they have to do with the `api.txt` generation?

Reply:
Yes, that's right, this library is used to extract the signature of the API for generating `api.txt` automatically